model-optimizer/extensions/middle/ConvertGroupedStridedSlice.py

```diff
@@ -30,6 +31,73 @@
 from mo.ops.unsqueeze import Unsqueeze
 
 
+def group_by_with_binary_predicate(xs: list, predicate: Callable) -> list:
+    """
+    It is an analogue of the function groupby from itertools, but with a binary predicate.
+    In other words, group_by_with_binary_predicate generates a break or new group every time
+    the value of the predicate function is False.
+    :param xs: list of grouped value
+    :param predicate: criterion of equality
+    :return: grouped list
+    """
+    if not xs:
+        return []
+    prev = xs[0]
+    sequence = [prev]
+    result = []
+    for x in xs[1:]:
+        if predicate(prev, x):
+            sequence.append(x)
+            prev = x
+        else:
+            result.append(sequence)
+            prev = x
+            sequence = [prev]
+    result.append(sequence)
+    return result
+
+
+def unique_by(xs: list, predicate: Callable) -> list:
+    """
+    This function groups elements of the list xs using 'predicate', and then takes one element from each group.
+    :param xs: input list
+    :param predicate: grouping criterion which is some binary predicate
+    :return: list with unique elements
+    """
+    groups = group_by_with_binary_predicate(xs, predicate)
+    result = []
+    for group in groups:
+        result.append(group[0])
+    return result
+
+
+default_attrs_of_strided_slice = {
+    'begin_mask' : int64_array([1]),
+    'end_mask': int64_array([1]),
+    'shrink_axis_mask': int64_array([0]),
+    'new_axis_mask': int64_array([0]),
+    'ellipsis_mask': int64_array([0])
+}
+
+
+def strided_slices_equality(lhs: Node, rhs: Node) -> bool:
+    """
+    Equality criterion for StridedSlice layers.
+    :param lhs: the first StridedSlice layer
+    :param rhs: the second StridedSlice layer
+    :return: True, if lhs and rhs have identical attributes 'slices', 'begin_mask', 'end_mask', 'ellipsis_mask',
+             'new_axis_mask', 'shrink_axis_mask', and False otherwise.
+    """
+    if not np.array_equal(lhs.slices, rhs.slices):
+        return False
+    for attr in ['begin_mask', 'end_mask', 'ellipsis_mask', 'new_axis_mask', 'shrink_axis_mask']:
```

Comment:
Is is possible that any of these attributes is not defined? I think that this should be an error. So may be we should add an assert in the constructor of the `StridedSlice` to check that these attributes are defined?

Reply:
According to https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/movement/StridedSlice_1.md, only attributes `begin_mask` and `end_mask` are required, other attributes (`ellipsis_mask`, `new_axis_mask`, `shrink_axis_mask`) are optional.

Reply:
The fact that these attributes are optional does not prevent MO to generate them always. If making them mandatory in the MO makes the MO code simpler let's do this!

Reply:
Done.